ruma-events/src/enums.rs

```diff
@@ -144,7 +128,7 @@ impl<'de> de::Deserialize<'de> for AnyEvent {
         D: de::Deserializer<'de>,
     {
         let json = Box::<RawJsonValue>::deserialize(deserializer)?;
-        let EventDeHelper { ev_type } = from_raw_json_value(&json)?;
+        let EventDeHelper { ev_type, state_key, event_id } = from_raw_json_value(&json)?;
```

Comment:
Nice! Though if you're doing this, you should probably get rid of `is_compatible` at the same time (known event_type strings don't take precedence over `event_id` / `state_key` being present in the event fields).

Reply:
Would adding a room_id to the `EventDeHelper` then solve differentiating `Ephemeral` and `Baisc` events?

Reply:
Yes! :)

Reply:
Now that I'm thinking about it: `EventDeHelper` should use `Option<IgnoredAny>` instead of the actual field types that will later be used. We only care whether the fields are present, not what their values are.

Reply:
Good thinking!